docs/next/src/components/AnchorHeading.tsx

```diff
@@ -59,4 +53,15 @@ const AnchorHeading: React.FC<AnchorHeadingProps> = ({
   );
 };
 
+export function getAnchorLinkFromHeadingContent(content: string): string {
+  return `#${content
+    .trim()
+    .toLowerCase()
+    // Remove special characters
+    .replace(/[^a-zA-Z0-9 ]/g, '')
+    .split(' ')
+    .filter(Boolean)
+    .join('-')}`;
+}
```

Comment:
Pulled out and simplified it a little bit. 

Note the behavior is slightly different for consecutive whitespaces (happy to revert if needed):
- Before: `Foo⎵⎵bar` -> `foo--bar`
- After: `Foo⎵⎵bar` -> `foo-bar`